components/camel-xj/src/main/java/org/apache/camel/component/xj/XJEndpoint.java

```diff
@@ -0,0 +1,527 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.xj;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import org.apache.camel.CamelContext;
+import org.apache.camel.Component;
+import org.apache.camel.Exchange;
+import org.apache.camel.api.management.ManagedAttribute;
+import org.apache.camel.api.management.ManagedOperation;
+import org.apache.camel.api.management.ManagedResource;
+import org.apache.camel.component.xslt.ResultHandlerFactory;
+import org.apache.camel.component.xslt.XsltOutput;
+import org.apache.camel.spi.*;
+import org.apache.camel.support.EndpointHelper;
+import org.apache.camel.support.ProcessorEndpoint;
+import org.apache.camel.support.service.ServiceHelper;
+import org.apache.camel.util.ObjectHelper;
+import org.xml.sax.EntityResolver;
+
+import javax.xml.transform.*;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+// todo?: This class is just a copy of XsltEndpoint because of minor but important different requirements. Refactoring of XsltComponent so that we can extend it?
+
+/**
+ * Transforms json/xml message back and forth using a XSLT.
+ */
+@ManagedResource(description = "Managed XJEndpoint")
+@UriEndpoint(firstVersion = "2.25.0", scheme = "xj", title = "XJ", syntax = "xj:resourceUri", producerOnly = true, label = "transformation")
+public class XJEndpoint extends ProcessorEndpoint {
+
+    public static final String SAXON_TRANSFORMER_FACTORY_CLASS_NAME = "net.sf.saxon.TransformerFactoryImpl";
+
+    private volatile boolean cacheCleared;
+    private volatile XsltBuilder xslt;
+    private Map<String, Object> parameters;
+
+    private JsonFactory jsonFactory = new JsonFactory();
+
+    @UriParam
+    @Metadata(required = true)
```

Comment:
It would be nice to have description in Metadata annotations

Reply:
done.